### ISSUE TITLE
[RocksJava] encoding issue with tests

### ISSUE THREAD

**Issue by pshareghi** [facebook/rocksdb#554]
I am using Centos 6 in a Docker file. When I run the following as part of my Docker file, it fails.
```
make rocksdbjavastatic
```
Unfortunately, I don't have time to replicate the error and paste it here.

The issue it complains about is ```unmappable character for encoding ASCII``` in some of the java test files (the files with Chinese characters IIRC). I was able to fix the issue locally by modifying the [java Makefile line 170](https://github.com/facebook/rocksdb/blob/master/java/Makefile#L170) to
```
javac -encoding UTF-8 -cp $(MAIN_CLASSES):$(JAVA_TESTCLASSPATH) -d $(TEST_CLASSES)\
```

**Comment by adamretter** [comment]
What is your system locale set to?

**Comment by fyrz** [comment]
@pshareghi are you using trunk or 3.9.1 ? Because i comitted an ascii compliant test with: https://github.com/facebook/rocksdb/commit/0d13bbe2721883907f2e0c6d53d25bef7786800b

**Comment by pshareghi** [comment]
@adamretter That is a good question. It was LANG=en_US.UTF-8 according to ```env```

@fyrz Yes, I am using 3.9.1 which is the stable version we use in production.

**Comment by fyrz** [comment]
@pshareghi can you verify if the issue is solved in 3.10.0 ?

**Comment by pshareghi** [comment]
Thanks, @fyrz. 3.10.fb compiles fine.
